File: lang/nodejs/test/fun/vac/algorithms/sorting/BubbleSortTest.js
"use strict"

const BubbleSort      = require("fun/vac/algorithms/sorting/BubbleSort")
const SequenceBuilder = require("fun/vac/util/SequenceBuilder")
const Sequences       = require("fun/vac/util/Sequences")
const TestRunner      = require("fun/vac/util/TestRunner")

const testBubbleSort = () => {
    let size = 32768

    const arr = new Array(size)
    SequenceBuilder.packRandom(arr)

    let checksum = Sequences.parityChecksum(arr)

    BubbleSort.sort(arr)

    if (Sequences.parityChecksum(arr) !== checksum) {
        return false
    }

    if (!Sequences.isSorted(arr)) {
        return false
    }

    return true
}

if (module === require.main) {
    TestRunner.parseTest(testBubbleSort())
}

File: lang/python/test/fun/vac/algorithms/sorting/BubbleSortTest.py
from fun.vac.algorithms.sorting import BubbleSort
from fun.vac.util import SequenceBuilder
from fun.vac.util import Sequences
from fun.vac.util import TestRunner

def testBubbleSort() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    BubbleSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum:
        return False

    return Sequences.isSorted(arr)

if __name__ == "__main__":
    TestRunner.parseTest(testBubbleSort())
